### PR TITLE
Document Support for username and password field while specifying git and hg material.

### PR DESCRIPTION
**Pull request by vrushaliwaykole**
**This should not be merged until 19.4.0 is merged as the contract won't be released until then.**

GoCD issue for reference: [#6132](https://github.com/gocd/gocd/issues/6132)

### CODE REVIEW

**Comment by tomzo** [comment]
This should work out of the box without code changes.

File: README.md [hunk 1]
```diff
@@ -530,6 +536,20 @@ All scm materials can have filter object:
   "shallow_clone": true
 }
 ```
+<a name="git-material"/>
+Since GoCD 19.4.0 onwards, you can specify username and password as such
```
Comment (arvindsv) [inline README.md]: @vrushaliwaykole Is this `you can` or `you need to`? Especially with `format_version: 5`. If it is, `you can`, then it's probably best to describe what happens if it's defined in both places, the URL and the attributes.

Comment (vrushaliwaykole) [reply README.md]: @arvindsv It should be `you need to` as we have decided to remove support of username and password from URL. I will update the documentation.

**Comment by vrushaliwaykole** [comment]
@tomzo @arvindsv I have updated the `Readme`. Also, added changes for documenting `encrypted_password` by default instead of `password`.

**Comment by bdpiprava** [comment]
@tomzo - we have updated the Readme(https://github.com/tomzo/gocd-json-config-plugin/pull/46/commits/9bab29720d846297aceaba5448a6bd294882c1bb) for branch attribute for Hg material. 

Attributes `username`, `encrypted_password/password` and `branch` are optional attributes. Users are advised to use them instead of specifying the same in URL

See issue https://github.com/gocd/gocd/issues/6298 for more details.

**Comment by bdpiprava** [comment]
@tomzo - I have added a note for `ambiguous credentials`. If it looks okay to you please create a release for the plugin. I will update the https://github.com/gocd/gocd to bundle the new release of the plugin.

**Pull request merged**